### PR TITLE
Goonify chemical retrieval (round two)

### PR DESCRIPTION
**Pull request by morrowwolf**
# About the pull request

This PR aims to do the following:
Replace the current PMC chem team with a corporate security chem team led by a researcher.
The researcher can call a PMC team for assistance so we have escalation if required. (admin approved)

This should allow for escalation of force for what should be a fairly cooperative situation. Weyland Yutani and the USCM are close friends in the Neroid sector. There is no reason this should blow up as much as it does but there should be some tension.

As always, this is a work in progress. Nothing is 100% final.

# Explain why it's good for the game

Read above.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>
I briefly tested some features of this.

</details>

# Changelog

:cl: Morrow
add: Chemical retrieval is now primarily corporate goons with a backup team of PMCs if required.
/:cl:

### CODE REVIEW

**Comment by Ticker175** [comment]
+1

**Review by sleepynecrons — CHANGES_REQUESTED** [review]
image

Feel like the beacon is a little flat compared to the other handhelds in-game. Tried making up a quick example to better show what I mean.

image

Also the first frame of the beacon_active animation has these two almost invisible brighter shades that none of the other frames have.

**Review by sleepynecrons — APPROVED** [review]
Problems I had with the sprites are fixed. Approved :D

File: code/game/objects/items/handheld_distress_beacon.dm [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+	if(active)
+		to_chat(user, "[src] is already active!")
+		return
+
+	for(var/client/C in GLOB.admins)
+		if((R_ADMIN|R_MOD) & C.admin_holder.rights)
+			playsound_client(C,'sound/effects/sos-morse-code.ogg',10)
+	message_admins("[key_name(usr)] has requested a PMC Distress Beacon! (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];ccmark=\ref[usr]'>Mark</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];distress_pmc=\ref[usr]'>SEND</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];ccdeny_pmc=\ref[usr]'>DENY</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];adminplayerobservejump=\ref[usr]'>JMP</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];CentcommReply=\ref[usr]'>RPLY</A>)")
+	to_chat(usr, SPAN_NOTICE("A distress beacon request has been sent to the USCSS Royce."))
```
Comment (harryob) [inline code/game/objects/items/handheld_distress_beacon.dm:37]:
```suggestion
	to_chat(user, SPAN_NOTICE("A distress beacon request has been sent to the USCSS Royce."))
```

File: code/game/objects/items/handheld_distress_beacon.dm [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	. = ..()
+
+	if(active)
+		to_chat(user, "[src] is already active!")
+		return
+
+	for(var/client/C in GLOB.admins)
+		if((R_ADMIN|R_MOD) & C.admin_holder.rights)
+			playsound_client(C,'sound/effects/sos-morse-code.ogg',10)
+	message_admins("[key_name(usr)] has requested a PMC Distress Beacon! (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];ccmark=\ref[usr]'>Mark</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];distress_pmc=\ref[usr]'>SEND</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];ccdeny_pmc=\ref[usr]'>DENY</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];adminplayerobservejump=\ref[usr]'>JMP</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];CentcommReply=\ref[usr]'>RPLY</A>)")
```
Comment (harryob) [inline code/game/objects/items/handheld_distress_beacon.dm:36]:
```suggestion
	message_admins("[key_name(user)] has requested a PMC Distress Beacon! (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];ccmark=\ref[user]'>Mark</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];distress_pmc=\ref[user]'>SEND</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];ccdeny_pmc=\ref[user]'>DENY</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];adminplayerobservejump=\ref[user]'>JMP</A>) (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];CentcommReply=\ref[user]'>RPLY</A>)")
```

File: code/datums/emergency_calls/pmc.dm [hunk 3]
```diff
@@ -99,13 +99,13 @@
...
+	objectives += "Assume at least 30 units are located within the department. If they can not make more that should be all. Cooperate with the onboard wey-yu researcher to ensure all who know the complete recipe are kept silenced with a contract of confidentiality. All humans who have ingested the chemical must be brought back dead or alive. Viral scan is required for any humans who is suspected of ingestion. Full termination of the department is authorized if they do not cooperate, but this should be avoided UNLESS ABSOLUTELY NECESSARY."
```
Comment (harryob) [inline code/datums/emergency_calls/pmc.dm:108]:
```suggestion
	objectives += "Assume at least 30 units are located within the department. If they can not make more that should be all. Cooperate with the onboard Wey-Yu researcher to ensure all who know the complete recipe are kept silenced with a contract of confidentiality. All humans who have ingested the chemical must be brought back dead or alive. Viral scan is required for any humans who is suspected of ingestion. Full termination of the department is authorized if they do not cooperate, but this should be avoided UNLESS ABSOLUTELY NECESSARY."
```

File: code/modules/admin/topic/topic.dm [hunk 4]
```diff
@@ -2001,6 +2012,14 @@
 	log_game("[key_name_admin(approver)] has sent a randomized distress beacon, requested by [key_name_admin(ref_person)]")
 	message_admins("[key_name_admin(approver)] has sent a randomized distress beacon, requested by [key_name_admin(ref_person)]")
 
+/datum/admins/proc/accept_PMC_ert(mob/approver, mob/ref_person)
```
Comment (harryob) [inline code/modules/admin/topic/topic.dm:2015]:
```suggestion
/datum/admins/proc/accept_pmc_ert(mob/approver, mob/ref_person)
```

File: code/modules/admin/topic/topic.dm [hunk 5]
```diff
@@ -1899,6 +1899,17 @@
 		addtimer(CALLBACK(src, PROC_REF(accept_ert), usr, locate(href_list["distress"])), 10 SECONDS)
 		//unanswered_distress -= ref_person
 
+	if(href_list["distress_pmc"]) //Wey-Yu specific PMC distress signal for chem retrieval ERT
+		distress_cancel = FALSE
+		message_admins("[key_name_admin(usr)] has opted to SEND the distress beacon! Launching in 10 seconds... (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];distresscancel=\ref[usr]'>CANCEL</A>)")
+		addtimer(CALLBACK(src, PROC_REF(accept_PMC_ert), usr, locate(href_list["distress"])), 10 SECONDS)
```
Comment (harryob) [inline code/modules/admin/topic/topic.dm:1905]:
```suggestion
		addtimer(CALLBACK(src, PROC_REF(accept_pmc_ert), usr, locate(href_list["distress"])), 10 SECONDS)
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**